backend/roles/role-managers/active-mixplay-users.js

```diff
@@ -0,0 +1,142 @@
+
+// Active user toggle
+let activeUserListStatus = settings.getActiveMixplayUserList() === false ? false : true;
+
+// User Timeout Settings
+let userInactiveTimeSetting = settings.getActiveMixplayUserListTimeout() != null ? settings.getActiveMixplayUserListTimeout() : 10;
+let inactiveTimer = userInactiveTimeSetting * 60000;
+
+// Timer and chatter list.
+let cycleActiveTimer = [];
+let activeMixplayUsers = [];
+
+function isUsernameActiveUser(username) {
+    let existingUserIndex = activeMixplayUsers.findIndex((obj => obj.username === username));
+    if (existingUserIndex !== -1) {
+        return true;
+    }
+    return false;
+}
+
+async function addOrUpdateActiveUser(user) {
+    if (!activeUserListStatus) {
+        return;
+    }
+
+    // Stop early if user shouldn't be in active chatter list.
+    let userDB = await userDatabase.getUserByUsername(user.username);
+    if (userDB.disableActiveUserList) {
+        logger.debug(userDB.username + " is set to not join the active mixplay user list.");
+        return;
+    }
+
+    // User should be okay, add them!
+    let date = new Date;
+    let currentTime = date.getTime();
+
+    let existingUserIndex = activeMixplayUsers.findIndex((obj => obj.userId === user.user_id));
+
+    // If user exists, update their time and stop.
+    if (existingUserIndex !== -1) {
+        logger.debug(user.username + " is still active in mixplay. Updating their time.");
+        activeMixplayUsers[existingUserIndex].time = currentTime;
+        return;
+    }
+
+    // Else, we're going to push the new user to the active array.
+    logger.debug(user.username + " has become active on mixplay. Adding them to active mixplay list.");
+    user = {
+        userId: user.userID,
+        username: user.username,
+        time: currentTime
+    };
+    activeMixplayUsers.push(user);
+}
+
+function removeLeavingUser(username) {
+    for (let userIndex in activeMixplayUsers) {
+        if (activeMixplayUsers[userIndex] != null) {
+            let user = activeMixplayUsers[userIndex];
+            if (user.username === username) {
+                logger.debug(user.username + " has left mixplay. Removing from active mixplay list.");
+                activeMixplayUsers.splice(userIndex, 1);
+            }
+        }
+    }
+}
+
+function clearInactiveUsers() {
+    logger.debug("Clearing inactive people from active mixplay users list.");
+    let date = new Date;
+    let currentTime = date.getTime();
+    let expiredTime = currentTime - inactiveTimer;
+    for (let userIndex in activeMixplayUsers) {
+        if (activeMixplayUsers[userIndex] != null) {
+            let user = activeMixplayUsers[userIndex];
+            if (user.time <= expiredTime) {
+                logger.debug(user.username + " has gone inactive on mixplay. Removing them from active mixplay list.");
+                activeMixplayUsers.splice(userIndex, 1);
+            }
+        }
+    }
+}
```

Comment:
This can be shortened to:
```js
function clearInactiveUsers() {
    logger.debug("Clearing inactive people from active mixplay users list.");
    let expiredTime = new Date().getTime() - inactiveTimer;
    activeMixplayUsers = activeMixplayUsers.filter(u => u != null && u.time <= expiredTime);
}
```